lib/AST/ASTContext.cpp

```diff
@@ -1103,8 +1103,29 @@ ProtocolDecl *ASTContext::getProtocol(KnownProtocolKind kind) const {
   // Find all of the declarations with this name in the appropriate module.
   SmallVector<ValueDecl *, 1> results;
 
-  const ModuleDecl *M;
+  const ModuleDecl *M = nullptr;
   switch (kind) {
+  // The Copyable protocol is not defined in any module. In particular, we can't
+  // define it in the stdlib because it's required to exist in order to
+  // typecheck all parts of the stdlib, so you would have to encounter the decl
+  // first. So we generate it on-demand and pretend it is in the stdlib.
```

Comment:
I'm not sure why it's any different from, eg, Sendable. What doesn't work if its declared in source?

Reply:
Ok, I see. The -parse-stdlib issue isn't a bootstrapping problem where you "haven't encountered the decl first", it's literally that test cases that use -parse-stdlib don't load the real stdlib, so they should define the dummy protocol. I still think that's cleaner than synthesizing it like this.

Reply:
Hm okay. I thought maybe `-parse-stdlib` was a more fundamentally required functionality of the compiler, but what you said makes it seem like it's mostly internally used for testing. So for now I'll go with the "don't add constraint if it's missing" since that's cleaner.

Reply:
actually we do build parts of the stdlib with `-parse-stdlib` so I'm going to keep this synthesizing hack.